src/ParallelExecutorFactory.php

```diff
@@ -337,7 +337,7 @@ private static function getNoopClosure(): Closure
 
     private static function getScriptPath(): string
     {
-        $pwd = $_SERVER['PWD'];
+        $pwd = $_SERVER['PWD'] ?? '';
```

Comment:
Why not try [`getcwd`](https://www.php.net/manual/en/function.getcwd.php)?

Reply:
does it work in this case? The trouble is I couldn't easily reproduce the issue so any "fix" is a random guess until someone tries it

Reply:
Hi!

It's easily reproducible in a Docker environment if noone has set `$PWD`, e.g.:
```
$ docker run -it --rm --entrypoint="" php:latest php -r "echo \$_SERVER['PWD'] . PHP_EOL;"

Warning: Undefined array key "PWD" in Command line code on line 1
```

This is the same for `docker exec` or `docker compose exec`.

The `$PWD` variable will be usually set though e.g. a shell, at least [POSIX-like shells should do that](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html#tag_08_03). That's why I've reset the [Docker PHP entrypoint](https://github.com/docker-library/php/blob/9efa150e7e28425da012816c607de298e22a946d/8.2/bookworm/cli/docker-php-entrypoint), otherwise the command is called via a shell script and the variable would be present there, e.g.:
```
$ docker run --rm php:latest php -r "echo \$_SERVER['PWD'] . PHP_EOL;"
/
```

This could do the trick:
```suggestion
        $pwd = $_SERVER['PWD'] ?? getcwd();
```

E.g.:
```
$ docker run --rm --entrypoint="" php:latest php -r "echo \$_SERVER['PWD'] ?? getcwd() . PHP_EOL;"
/
```